### PR TITLE
Expand tests for service accounts [OSD-9355]

### PR DESCRIPTION
**Pull request by aliceh**
Unit tests are expanded to cover the various scenarios around service accounts and privileged namespaces

### CODE REVIEW

File: pkg/webhooks/namespace/namespace_test.go [hunk 1]
```diff
@@ -360,6 +360,52 @@ func TestLayeredProducts(t *testing.T) {
...
+			targetNamespace: "kube-system",
+			username:        "system:serviceaccounts:openshift-test-ns",
+			userGroups:      []string{"system:serviceaccounts:openshift-test-ns", "system:authenticated", "system:authenticated:oauth"},
+			operation:       admissionv1.Create,
+			shouldBeAllowed: true,
+		},
+		{
+			// serviceaccounts in privileged namespaces can interact with privileged namespaces
+			testID:          "sa-create-priv-ns",
+			targetNamespace: "redhat-user",
```
Comment (sam-nguyen7) [inline pkg/webhooks/namespace/namespace_test.go:394]: is "redhat-user" an actually namespace? or is this testing redhat-* namespaces?

Comment (sam-nguyen7) [inline pkg/webhooks/namespace/namespace_test.go:394]: is "redhat-user" an actually namespace? or is this testing redhat-* namespaces?

File: pkg/webhooks/namespace/namespace_test.go [hunk 2]
```diff
@@ -399,6 +445,42 @@ func TestServiceAccounts(t *testing.T) {
...
+			operation:       admissionv1.Create,
+			shouldBeAllowed: false,
+		},
+		{
+			// serviceaccounts in unprivileged namespaces can not interact with privileged namespaces
+			testID:          "sa-create-priv-ns",
+			targetNamespace: "redhat-user",
+			username:        "system:serviceaccounts:unpriv-ns",
+			userGroups:      []string{"system:serviceaccounts:unpriv-ns", "system:authenticated", "system:authenticated:oauth"},
+			operation:       admissionv1.Create,
```
Comment (sam-nguyen7) [inline pkg/webhooks/namespace/namespace_test.go:481]: is Create the only verb/operation we care about?

**Comment by aliceh** [comment]
"redhat-user" is a totally made up  redhat-* namespace :-)

**Comment by aliceh** [comment]
> is Create the only verb/operation we care about?

good question. I should probably add more tests with other verbs.

**Pull request merged**